Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java

```diff
@@ -56,9 +57,35 @@ public boolean handleOptionSelection(int itemId)
 				return true;
 
 			case R.id.menu_refresh:
+				mView.showScanDialog();
 				GameDatabase databaseHelper = DolphinApplication.databaseHelper;
-				databaseHelper.scanLibrary(databaseHelper.getWritableDatabase());
```

Comment:
I wonder will the old helper still be called somewhere at all?

Reply:
yes, it is called inside the GameProvider content provider, which is by nature should be a blocking call, I can reuse the other one with Observable but it looks an overkill for other cases.